python/ray/data/tests/test_logging.py

```diff
@@ -148,20 +150,22 @@ def test_json_logging_configuration(
         log_contents = file.read()
 
     # Validate the log is in JSON format (a basic check for JSON)
-    assert all(
-        log_line.startswith("{") and log_line.endswith("}")
-        for log_line in log_contents.splitlines()
-    )
+    messages = set()
+    for log_line in log_contents.splitlines():
+        log_dict = json.loads(log_line)  # will error if not a json line
+        messages.add(log_dict["message"])
 
-    assert '"message": "ham"' in log_contents
-    assert '"message": "turkey"' in log_contents
+    assert "ham" in messages
+    assert "turkey" in messages
 
     # Validate console logs are in text mode
     console_log_output = capsys.readouterr().err
-    assert not any(
-        log_line.startswith("{") and log_line.endswith("}")
-        for log_line in console_log_output.splitlines()
-    )
+    for log_line in console_log_output.splitlines():
+        try:
+            json.loads(log_line)
+            raise ValueError("Expected log line to be logged without JSON formatting.")
+        except json.JSONDecodeError:
+            pass
```

Comment:
nit: can use `pytest.raises` to clean this up:
```suggestion
        with pytest.raises(json.JSONDecodeError):
            json.loads(log_line)
```